scripts/functions/utility

```diff
@@ -60,10 +60,7 @@ __rvm_remove_from_path()
   if
     [[ "$PATH" == *"//"* || "$PATH" == *"/:"* ]]
   then
-    if [[ "${_system_type}" == "Darwin" ]]
```

Comment:
the `-E` flag might not be supported on some systems, it might be safer to use here `[[ "$(command uname)" == "Darwin" ]]` - it will be slower, but more reliable

Reply:
Yeah, it seems like this was why the original bug was happening actually.

According to the discussion in #4618, it was actually the other way around `-r` is not supported on some systems while `-E` should be in most modern versions of `sed` since the POSIX.2 standard.

I think @clive-hetzner summed it up well when he commented:

> I feel that altering the sed behaviour is just (re)hiding the underlying bug that system detection is never executed in this one case - re-entering an rvm folder after having exited it.
> There are other source paths (routes) through the code that could be impacted by the missing system detection in this case.

So the underlying issue he mentioned was that `_system_type` was not set in this case (changing directory to a ruby project).  Therefore the original code assumed `sed -r` was supported and resulted in error:

```
sed: illegal option -- r
```

The other thing we discussed was that the original test is based on platform which is conflated to mean `sed` version.  So the underlying assumption is that `-r` vs. `-E` support is tied to `sed` version & platform.  That assumption turns out to be incorrect, as various versions of macOS have [included various versions of `sed`][1] throughout the years.  Ideally we could detect the version of `sed` or flags supported, but that might be difficult to get right everywhere.

So this proposed solution simplifies things a bit and makes the assumption that POSIX.2 is supported by all the modern *nix platforms.  Not sure if it's the "best" way or if there is one way to support all platforms well without extensive cross-platform testing.

[1]: https://github.com/rvm/rvm/issues/4618#issuecomment-501001444

Reply:
So to simplify & summarize the problem a bit, there are actually 2 issues:

1. Version / variant of `sed` used across *nix platforms is different, and thus supports different flags.  (POSIX standards could help us out here)
2. Platform detection wasn't being executed properly & could crop up elsewhere

Reply:
Another alternate idea I just tested to sidestep the issue could be to just use plain old POSIX Basic RegEx (BRE) instead of Extended RegEx (ERE):

```
PATH="$(\sed -e 's/\/\{1,\}/\//g' -e 's/\/:/:/' <<<$PATH)"
```

This should work on all POSIX platforms and even older versions of `sed`, because it's just using the [`"\{m,n\}"` BRE syntax][1] for matching one or more slashes.  It also needs to escape the literal forward slashes `/` due to the default BRE separator being forward slash `/`.  It's a bit more cumbersome to look at which just shows the readability advantages of using [ERE features][2] such as `+` and alternate find/replace separators.  Yet, it's more cross-platform capable using basic POSIX [`sed`][3] features and solves the issue! 😄

@mpapis:  I'd love to hear what you think of this alternate solution, and the other ideas mentioned above.

[1]: http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html#tag_09_03_06
[2]: http://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap09.html#tag_09_04
[3]: http://pubs.opengroup.org/onlinepubs/9699919799/utilities/sed.html